File: fastApiProject/entity/GuessWord.py
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime


class GuessWord(BaseModel):
    """猜字游戏目标字实体"""
    id: Optional[int] = Field(None, description="目标字唯一ID")
    word: str = Field(..., description="要猜的字/词")
    hint: Optional[str] = Field(None, description="提示信息")
    difficulty: int = Field(1, description="难度等级:1-简单,2-中等,3-困难")
    is_passed: bool = Field(False, description="是否已通过")
    pass_count: int = Field(0, description="通过次数")
    created_at: Optional[datetime] = Field(None, description="创建时间")
    embedding: Optional[List[float]] = Field(None, description="词语向量数据")

    class Config:
        from_attributes = True

File: blog-agent/agent/src/main/java/com/xiafan/agent/entity/GuessWord.java
package com.xiafan.agent.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.xiafan.agent.repository.PostgresJsonTypeHandler;
import lombok.Data;
import lombok.Getter;

import java.time.LocalDateTime;
import java.util.List;

/** guess_words */
@TableName(value = "guess_words", autoResultMap = true)
@Data
public class GuessWord {
    @TableId(value = "id", type = IdType.AUTO)
    private Integer id;
    @TableField("word")
    private String word;
    @TableField("hint")
    private String hint;
    @TableField("difficulty")
    private Integer difficulty = 1;

    @TableField("is_passed")
    @Getter(onMethod_ = @JsonProperty("is_passed"))
    private boolean passed = false;

    @TableField("pass_count")
    private Integer passCount = 0;
    @TableField("created_at")
    private LocalDateTime createdAt;
    @TableField(value = "embedding", typeHandler = PostgresJsonTypeHandler.class)
    private List<Double> embedding;
}
